### PR TITLE
feature: du shared extents reflink

### PR DESCRIPTION
**Pull request by cre4ture**
Addresses #3906:

On copy-on-write filesystems it is possible to copy files without really copying the contents. The contents are copied when modified.

This pullrequests introduces a new feature for "du --shared-extents" that consideres the shared extents of the files, not counting it double.

### CODE REVIEW

File: .vscode/cSpell.json [hunk 1]
```diff
@@ -2,20 +2,38 @@
 {
   // version of the setting file
   "version": "0.2",
-
   // spelling language
   "language": "en",
-
   // custom dictionaries
-  "dictionaries": ["acronyms+names", "jargon", "people", "shell", "workspace"],
+  "dictionaries": [
```
Comment (sylvestre) [inline .vscode/cSpell.json:8]: please move the unrelated changes into a different PR

**Review by sylvestre — CHANGES_REQUESTED** [review]
As said in the ticket, i don't want to add new features that GNU doesn't have until we are at parity. Sorry.

**Comment by cre4ture** [comment]
@sylvestre please approve the workflows. I'd like to see how my changes are accepted on the other systems.

> As said in the ticket, i don't want to add new features that GNU doesn't have until we are at parity. Sorry.

I see. So what do we want to do with this pull request? I've seen you started commented already. Thanks for this :-) . But if its not gona be merged, we should stop working on it. Whats your plan?

File: .vscode/cSpell.json (hunk 1 above)
Comment (cre4ture) [reply .vscode/cSpell.json:8]: the format and sorting changes in the subfolder ".vscode" where done by vscode itself. So I was assuming that this is fine. If you don't accept that, this means for me that I have to do the changes there manually which is inconvenient. I guess I'm not the only person with this issue, am I? How do you do the changes there?

**Comment by sylvestre** [comment]
no activity for a while, closing
